### DISCUSSION TITLE
How does send_keys impact subsequent steps like find?

### DISCUSSION THREAD

**Discussion by MMore** [keathley/wallaby#673] [Q&A] [answered]
Hello,
I’ve just started playing around with wallaby. I like the DSL so far. But I have an issue I don’t really understand.
```
      session
      |> visit(
        "https://id.atlassian.com/login"
      )
      |> fill_in(text_field("username"), with: "email@example.org")
      |> send_keys([:enter])
      |> find(css("#username"))
```
-> I get `** (Wallaby.QueryError) Expected to find 1, visible element that matched the css '#username' but 0, visible elements were found`.

When I execute the same without the line `|> send_keys([:enter])`, it finds the #username field. According to the typespecs `send_keys/2,3` returns the `parent`, which should be the session. What am I doing wrong? The field is visible and I can see it (headless chrome disabled).

**Comment by mhanberg** [comment, marked as answer]
Hi, the reason why you are seeing this is due to the way the Atlassian login page works.

After you fill in your email, the username input element with id username `#username` disappears and is replaced with a span. Here is a screenshot that shows the DOM.

<img width="2032" alt="image" src="https://user-images.githubusercontent.com/5523984/158925355-06e68913-cc72-4c4b-ab35-fbb8c985c9cb.png">

**Reply by MMore** [reply]
Ah yeah, you're right I assumed falsely that this element stays like it was before. Don't do those things at night ;) Thank you!
